app/main.py

```diff
@@ -1,6 +1,17 @@
 from typing import Callable
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cached_results = {}
+
+    @wraps(func)
+    def inner(*args: list) -> list:
```

Comment:
Using `list` as the type hint for `*args` is misleading because `*args` is a tuple, not a list. Consider changing the type hint to `tuple`.